SQL_Assignment.txt

```diff
@@ -0,0 +1,1093 @@
++------+------------+----------+
+21) List all the largest orders for October 3, for each salesperson.
+
+22) Find all customers located in cities where Serres has customers.
+
+mysql> select CNAME,CITY from Customers where CITY in(select CITY from Salespeople where SNUM=1002);
++----------+---------+
+| CNAME    | CITY    |
++----------+---------+
+| Liu      | Sanjose |
+| Cisneros | SanJose |
++----------+---------+
+
+23) Select all customers with a rating above 200.
+
+mysql> select * from Customers where RATING>200;
++------+----------+---------+--------+------+
+| CNUM | CNAME    | CITY    | RATING | SNUM |
++------+----------+---------+--------+------+
+| 2004 | Grass    | Berlin  |    300 | 1002 |
+| 2008 | Cisneros | SanJose |    300 | 1007 |
++------+----------+---------+--------+------+
+
+24) Count the number of salespeople currently having orders in the orders table.
+
+mysql> select count(distinct SNUM) from Orders;
++----------------------+
+| count(distinct SNUM) |
++----------------------+
+|                    5 |
++----------------------+
+
+25) Write a query that produces all customers serviced by salespeople with a commission above 12%. Output the customer’s name,
+salesperson’s name and the salesperson’s rate of commission.
+
+mysql>  select c.CNAME,s.COMM from Customers c join Salespeople s on c.SNUM=s.SNUM and s.COMM>12;
++----------+------+
+| CNAME    | COMM |
++----------+------+
+| Liu      |   13 |
+| Grass    |   13 |
+| Cisneros |   15 |
++----------+------+
+
+26) Find salespeople who have multiple customers.
+
+mysql> select SNUM,count(SNUM) from Customers group by SNUM having count(SNUM)>1;
++------+-------------+
+| SNUM | count(SNUM) |
++------+-------------+
+| 1001 |           2 |
+| 1002 |           2 |
++------+-------------+
+
+27) Find salespeople with customers located in their own cities.
+
+mysql> select s.SNAME,c.CNAME,s.CITY from Salespeople s join Customers c on s.CITY=c.CITY Order by CITY;
++--------+----------+---------+
+| SNAME  | CNAME    | CITY    |
++--------+----------+---------+
+| Motika | Hoffman  | Londan  |
+| Fran   | Hoffman  | Londan  |
+| Peel   | Hoffman  | Londan  |
+| Serres | Cisneros | SanJose |
+| Serres | Liu      | SanJose |
++--------+----------+---------+
+
+28) Find all salespeople whose name starts with ‘P’ and fourth character is ‘I’.
+
+mysql> select SNAME from Salespeople where SNAME like 'P__L%';
++-------+
+| SNAME |
++-------+
+| Peel  |
++-------+
+
+29) Write a query that uses a subquery to obtain all orders for the customer named ‘Cisneros’. Assume you do not know his customer number.
+
+mysql> select ONUM from Orders  where SNUM=(select SNUM from Customers where CNAME='Cisneros');
++------+
+| ONUM |
++------+
+| 3001 |
+| 3006 |
++------+
+30) Find the largest orders for Serres and Rifkin.
+
+mysql> select SNUM,max(AMT) from Orders group by SNUM  having SNUM in(select SNUM from Salespeople where SNAME='Serres' or SNAME='Rifkin');

```

Comment:
Display the salesperson's name in this query. It helps to see which order belongs to which salesperson.